control/strategy/pool_test.go

```diff
@@ -0,0 +1,207 @@
+	Convey("Given available collector type plugin", t, func() {
+		plg := NewMockAvailablePlugin()
+		Convey("When new plugin pool is being created with expected plugin type", func() {
+			tcs := []struct {
+				PlgType       plugin.PluginType
+				Strategy      plugin.RoutingStrategyType
+				Concurrency   int
+				Subscriptions int
+				Exclusiveness bool
+				Expected      bool
+			}{
+				// type, strategy, concurrency count, number of subscriptions, exclusiveness, eligibility
+				{plugin.CollectorPluginType, plugin.DefaultRouting, 1, 0, false, false},
+				{plugin.CollectorPluginType, plugin.DefaultRouting, 1, 1, false, false},
+				{plugin.CollectorPluginType, plugin.DefaultRouting, 2, 1, false, false},
+				{plugin.CollectorPluginType, plugin.DefaultRouting, 3, 1, false, false},
+				{plugin.CollectorPluginType, plugin.DefaultRouting, 1, 2, false, true},
+				{plugin.CollectorPluginType, plugin.DefaultRouting, 1, 3, false, true},
+				{plugin.CollectorPluginType, plugin.DefaultRouting, 3, 3, false, false},
+				{plugin.CollectorPluginType, plugin.DefaultRouting, 1, 3, true, false},
+
+				{plugin.CollectorPluginType, plugin.StickyRouting, 999, 0, false, false},
+				{plugin.CollectorPluginType, plugin.StickyRouting, 999, 1, false, false},
+				{plugin.CollectorPluginType, plugin.StickyRouting, 999, 2, false, true},
+				{plugin.CollectorPluginType, plugin.StickyRouting, 999, 3, false, true},
+				{plugin.CollectorPluginType, plugin.StickyRouting, 999, 4, false, false},
+				{plugin.CollectorPluginType, plugin.StickyRouting, 999, 2, true, false},
+
+				{plugin.CollectorPluginType, plugin.ConfigRouting, 999, 0, false, false},
+				{plugin.CollectorPluginType, plugin.ConfigRouting, 999, 1, false, false},
+				{plugin.CollectorPluginType, plugin.ConfigRouting, 999, 2, false, false},
+				{plugin.CollectorPluginType, plugin.ConfigRouting, 999, 3, false, false},
+				{plugin.CollectorPluginType, plugin.ConfigRouting, 999, 4, false, true},
+				{plugin.CollectorPluginType, plugin.ConfigRouting, 999, 5, false, true},
+				{plugin.CollectorPluginType, plugin.ConfigRouting, 999, 4, true, false},
+			}
+			Convey("Then new pool eligibility is defined", func() {
+				for i, tc := range tcs {
+					plg.WithPluginType(tc.PlgType).
+						WithStrategy(tc.Strategy).
+						WithExclusive(tc.Exclusiveness).
+						WithConCount(tc.Concurrency).
+						WithID(uint32(i))
+
+					pool, _ := NewPool(plg.String(), plg)
+
+					for j := 0; j < tc.Subscriptions; j++ {
+						pool.Subscribe(strconv.Itoa(j), BoundSubscriptionType)
+					}
+
+					Convey(fmt.Sprintf(
+						"{strategy = %s, concurreny = %d, subscriptions = %d, exclusiveness = %v, count = %d}",
+						tc.Strategy.String(),
+						tc.Concurrency,
+						tc.Subscriptions,
+						tc.Exclusiveness,
+						pool.Count(),
+					),
+						func() {
+							So(pool.SubscriptionCount(), ShouldEqual, tc.Subscriptions)
+							So(pool.Eligible(), ShouldEqual, tc.Expected)
+						})
+				}
+			})
+		})
+	})
+}
```

Comment:
So do I; we really need to get this pattern into the other tests we write in order to implement a real set of **small** tests as part of the **Testing Redux** epic; great job @marcin-krolik 👍